src/modules/job-manager/queue.c

```diff
@@ -547,12 +576,61 @@ static void queue_enable_cb (flux_t *h,
         flux_log_error (h, "error responding to job-manager.queue-enable");
 }
 
-static void queue_stop_allocations (struct queue *queue) {
-    if (alloc_pending_count (queue->ctx->alloc) > 0) {
+int job_in_queue (struct queue *queue,
+                  struct job *job,
+                  const char *name,
+                  bool *in_queue)
+{
+    const char *qname = NULL;
+    if (!name) {
+        (*in_queue) = true;
+        return 0;
+    }
+    if (job_queue_name (queue, job, &qname) < 0)
+        return -1;
+    (*in_queue) = qname && !strcmp (qname, name) ? true : false;
+    return 0;
+}
+
+static int queue_start_allocations (struct queue *queue, const char *name)
+{
+    struct job *job = zhashx_first (queue->ctx->active_jobs);
+    while (job) {
+        bool in_queue;
+        if (job_in_queue (queue, job, name, &in_queue) < 0)
+            goto next;
+        if (in_queue) {
+            if (!job->alloc_queued
+                && !job->alloc_pending
+                && job->state == FLUX_JOB_STATE_SCHED) {
+                /* XXX or event_job_action()? */
+                if (alloc_enqueue_alloc_request (queue->ctx->alloc, job) < 0)
+                    return -1;
+                if (alloc_queue_recalc_pending (queue->ctx->alloc) < 0)
+                    return -1;
+            }
+        }
+    next:
+        job = zhashx_next (queue->ctx->active_jobs);
+    }
+    return 0;
+}
+
+static void queue_stop_allocations (struct queue *queue, const char *name) {
```

Comment:
Should this and the previous function just be called `queue_stop()` and `queue_start()`?

Also should be a line break before the opening brace.

Reply:
hmmm there's a `queue_start_all()` function already, which was based on `queue_enable_all()`.  Perhaps `queue_enable_all()` probably should be `jobq_enable_all()`, which follows `jobq_enable()`, then the code for `queue_start` follows after that.